### PR TITLE
[4.0] FIX Unreliable way of loading ServiceProviderInterface class in ExtensionManagerTrait.php

### PR DESCRIPTION
**Pull request by joeforjoomla**
Pull Request for Issue #22825 .

### Summary of Changes
Use the Factory to cache loaded extensions throughout different Application instances

### Testing Instructions
See previous PR https://github.com/joomla/joomla-cms/pull/22825

### Expected result
Loaded extensions from the cache are always correct

### Actual result
Loaded extensions from the cache are always correct

### CODE REVIEW

**Comment by laoneo** [comment]
If you want to make a cache, then better to use the ExtensionHelper class inszead of the already polluted Factory.

**Comment by joeforjoomla** [comment]
I agree @laoneo this PR has been modified to properly implement an extensions cache using the ExtensionHelper class. 
Please test.

File: libraries/src/Extension/ExtensionHelper.php [hunk 1]
```diff
@@ -19,6 +19,14 @@
  */
 class ExtensionHelper
 {
+	/**
+	 * Global loaded extensions through the container
+	 *
+	 * @var    CMSApplicationInterface
```
Comment (laoneo) [inline libraries/src/Extension/ExtensionHelper.php]: This type here is wrong

**Review by laoneo — CHANGES_REQUESTED** [review]

File: libraries/src/Extension/ExtensionHelper.php (hunk 1 above)
Comment (joeforjoomla) [reply libraries/src/Extension/ExtensionHelper.php]: Fixed type to array.

**Review by laoneo — APPROVED** [review]

**Comment by joeforjoomla** [comment]
Can this PR be tested and merged ASAP? Otherwise it's difficult to fully test extensions everytime that a new alpha version is released. Pinging @wilsonge here, thanks George!

**Comment by wilsonge** [comment]
Thanks for the ping - sorry I missed this - LGTM!

**Comment by joeforjoomla** [comment]
Thanks @wilsonge ! You are the best :)

**Pull request merged**